node_modules/oae-principals/lib/api.user.js

```diff
@@ -119,63 +126,275 @@ var createUser = module.exports.createUser = function(ctx, tenantAlias, displayN
-
-    var values = {
-        'tenantAlias': tenantAlias,
-        'displayName': displayName,
+    var user = new User(tenantAlias, id, displayName, {
         'visibility': opts.visibility,
         'email': opts.email,
-        'emailPreference': opts.emailPreference,
         'locale': opts.locale,
         'publicAlias': opts.publicAlias,
+        'emailPreference': opts.emailPreference,
+        'emailVerified': opts.emailVerified,
         'smallPictureUri': opts.smallPictureUri,
         'mediumPictureUri': opts.mediumPictureUri,
-        'largePictureUri': opts.lagePictureUri
-    };
+        'largePictureUri': opts.lagePictureUri,
+        'acceptedTC': 0,
+        'guest': opts.guest
+    });
 
     // We store the timestamp at which the user accepted the Terms and Conditions
     // This allows for allowing users to re-accept the Terms and Conditions after they have been updated
-    var tcEnabled = PrincipalsConfig.getValue(tenantAlias, 'termsAndConditions', 'enabled');
-    if (tcEnabled && opts.acceptedTC) {
-        // Store the acceptedTC on the return user object
-        opts.acceptedTC = Date.now();
+    user.needsToAcceptTC = PrincipalsConfig.getValue(tenantAlias, 'termsAndConditions', 'enabled');
+    if (user.needsToAcceptTC && opts.acceptedTC) {
+        user.acceptedTC = Date.now();
+        user.needsToAcceptTC = false;
+    }
 
-        // Also persist it, but it must be a String because that is the column data-type in Cassandra
-        values.acceptedTC = opts.acceptedTC.toString();
+    PrincipalsDAO.createUser(user, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Emit an event so the user can be indexed by search
+        PrincipalsEmitter.emit(PrincipalsConstants.events.CREATED_USER, ctx, user);
+
+        // Return to the caller
+        callback(null, user);
+
+        // If the user information didn't come from an authoritative source
+        // we ask the user to verify his email
+        if (!isAdmin && user.email && !user.emailVerified) {
+            sendEmailToken(ctx, user);
+        }
+    });
+};
+
+/**
+ * Create a guest user account for an email address. If an existing (guest) user account
+ * was already associated with the email addres, it will be returned. If not, a guest
+ * account will be created.
+ *
+ * @param  {Context}    ctx                         Standard context object containing the current user and the current tenant
+ * @param  {String}     email                       The email for the user
+ * @param  {Function}   callback                    Standard callback function
+ * @param  {Object}     callback.err                An error that occurred, if any
+ * @param  {User}       callback.user               The user
+ * @param  {Boolean}    callback.isGuest            `true` if the returned user is a guest user
+ */
+var createGuestUser = module.exports.createGuestUser = function(ctx, email, callback) {
```

Comment:
This method name isn't entirely accurate either. Should probably be something like `getOrCreateGuestUser` instead